llama_index/query_engine/sql_vector_query_engine.py

```diff
@@ -0,0 +1,311 @@
+    "Original question: {query_str}\n"
+    "SQL query: {sql_query_str}\n"
+    "SQL response: {sql_response_str}\n"
+    "Transformed vector store query: {vector_store_query_str}\n"
+    "Vector store response: {vector_store_response_str}\n"
+    "Response: "
+)
+DEFAULT_SQL_VECTOR_SYNTHESIS_PROMPT = Prompt(DEFAULT_SQL_VECTOR_SYNTHESIS_PROMPT_TMPL)
+
+
+DEFAULT_SQL_AUGMENT_TRANSFORM_PROMPT_TMPL = (
+    "The original question is given below.\n"
+    "This question has been translated into a SQL query. Both the SQL "
+    "query and the response are given below.\n"
+    "The SQL response either answers the question, or should provide additional context that can be used "
+    "to make the question more specific.\n"
+    "Your job is to come up with a more specific question that needs to be answered "
+    "to fully answer the original question, "
+    "or 'None' if the original question has already been fully answered from the "
+    "SQL response. Do not create a new question "
+    "that is irrelevant to the original question; in that case return None instead. \n\n"
+    "Examples:\n\n"
+    "Original question: Please give more details about the demographics of the city with the highest "
+    "population. \n"
+    "SQL query: SELECT city, population FROM cities ORDER BY population DESC LIMIT 1\n"
+    "SQL response: The city with the highest population is New York City.\n"
+    "New question: Can you tell me more about the demographics of New York City?\n\n"
+    "Original question: Please compare the sports environment of cities in North America.\n"
+    "SQL query: SELECT city_name FROM cities WHERE continent = 'North America' LIMIT 3\n"
+    "SQL response: The cities in North America are New York, San Francisco, and Toronto.\n"
+    "New question: What sports are played in New York, San Francisco, and Toronto?\n\n"
+    "Original question: What is the city with the highest population?\n"
+    "SQL query: SELECT city, population FROM cities ORDER BY population DESC LIMIT 1\n"
+    "SQL response: The city with the highest population is New York City.\n"
+    "New question: None\n\n"
+    "Original question: What countries are the top 3 ATP players from?\n"
+    "SQL query: SELECT country FROM players WHERE rank <= 3\n"
+    "SQL response: The top 3 ATP players are from Serbia, Russia, and Spain.\n"
+    "New question: None\n\n"
+    "Original question: {query_str}\n"
+    "SQL query: {sql_query_str}\n"
+    "SQL response: {sql_response_str}\n"
+    "New question: "
+)
+DEFAULT_SQL_AUGMENT_TRANSFORM_PROMPT = Prompt(DEFAULT_SQL_AUGMENT_TRANSFORM_PROMPT_TMPL)
+
+
+def _default_check_stop(query_bundle: QueryBundle) -> bool:
+    """Default check stop function."""
+    return query_bundle.query_str.lower() == "none"
+
+
+def _format_sql_query(sql_query: str) -> str:
+    """Format SQL query."""
+    return sql_query.replace("\n", " ").replace("\t", " ")
+
+
+class SQLAugmentQueryTransform(BaseQueryTransform):
```

Comment:
just a thought: not sure how much value our query transform abstraction really adds. It's really just a (str -> str) function most of the time (maybe except HyDE)

Reply:
yeah...either way though we'd need to define either a function or class to run on queries though right